components/script/body.rs

```diff
@@ -40,25 +40,39 @@ use js::rust::wrappers::JS_ParseJSON;
 use js::rust::HandleValue;
 use js::typedarray::{ArrayBuffer, CreateWith};
 use mime::{self, Mime};
-use net_traits::request::{BodyChunkRequest, BodySource, RequestBody};
+use net_traits::request::{BodyChunkRequest, BodySource as NetBodySource, RequestBody};
 use script_traits::serializable::BlobImpl;
 use std::ptr;
 use std::rc::Rc;
 use std::str;
 use url::form_urlencoded;
 
+/// The Dom object, or ReadableStream, that is the source of a body.
+/// <https://fetch.spec.whatwg.org/#concept-body-source>
+#[derive(Clone)]
+pub enum BodySource {
+    /// A ReadableStream comes with a null-source.
+    Null,
+    /// Another Dom object as source,
+    /// TODO: store the actual object
+    /// and re-exctact a stream on re-direct.
```

Comment:
Please file an issue for this if there isn't one already.

Reply:
Yes that would be https://github.com/servo/servo/issues/26686